src/ripple/app/misc/Validations.cpp

```diff
@@ -387,6 +387,35 @@ class ValidationsImp : public Validations
         return ret;
     }
 
+    std::uint32_t getValidationTime (uint256 const& hash, int minValidations)
+    {
+        if (minValidations < 1)
+            minValidations = 1;
+
+        std::vector <std::uint32_t> times;
+
+        {
+            ScopedLockType sl (mLock);
+
+            auto j = findSet (hash);
+            for (auto& it : *j)
+            {
+                if (it.second->isTrusted())
+                    times.push_back (it.second->getSignTime());
+            }
+        }
+
+        if (times.size() < minValidations)
+            return 0;
+
+        if (times.size() == 1)
+            return times[0];
+
+        // Return median signing time
+        std::sort (times.begin (), times.end ());
+        return (times[times.size() / 2] + times[(times.size()-1) / 2]) / 2;
```

Comment:
You don't want one validator that's a decade off to make the number meaningless. We could sanity check it and use the close time if the signing time isn't within the close resolution, but I'm not sure it's worth it.